### PR TITLE
[Quickstart] Added validation to password question

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

**Error:**
```
[Set up your database access details]
? Database User Username [Press Enter to use 'Quickstart-1621862864']: Quickstart-1621862864
? Database User Password [Press Enter to use an auto-generated password 'VVdJTGZdUklX']: password
Error: POST https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/databaseUsers: 400 (request "COMMON_PASSWORD") The password provided is too weak, as it can be found in most commonly used password lists.
```

**Fix:**
```
[Set up your database access details]
? Database User Username [Press Enter to use 'Quickstart-1621866516']: Quickstart-1621866516
X Sorry, your reply was invalid: the password provided is too weak, as it can be found in most commonly used password lists: password
? Database User Password [Press Enter to use an auto-generated password 'RWV1RVlnVEto']: password!21212!!!

```

### CODE REVIEW

File: internal/validate/common_password.go
Comment (andreaangiolillo) [inline internal/validate/common_password.go:17]: here I am using the same list used by the endpoint. I am not happy about this list given that we need to keep it updated with the one in the endpoint. I am happy to consider another way to validate this

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:164]: [q] tests?

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:162]: [nit] this is two long, for sentinel errors you want them short
I would make it (but feel free to improve on)
```go
var ErrWeakPassoword = errors.New("invalid password")

func WeakPassword(val interface{}) error {
...
if commonPasswords[password] {
	return fmt.Errorf("%w: password is too common", ErrWeakPassoword)
}
```
don't return the password when you asked it masked just to expose it in an error

File: internal/validate/common_password.go
Comment (threebee) [reply internal/validate/common_password.go:17]: Which endpoint are you talking about? Could you please point out the endpoint?

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:172]: what about checking length? 8 is standard but now days is considered insecure (can be cracked in a couple hours) 10 may be a good value

File: internal/validate/common_password.go
Comment (threebee) [reply internal/validate/common_password.go:17]: Instead of looking up for weak password, could we add a password policy? Something like atleast 2 uppercase, 2 lowercase, 1 number and 1 special character?

File: internal/validate/common_password.go
Comment (gssbzn) [inline internal/validate/common_password.go:413]:
```suggestion
	"zzzzzz":      true,
}
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
good add here but main thing before getting it merge, tests

File: internal/validate/common_password.go
Comment (gssbzn) [reply internal/validate/common_password.go:17]: could we make that a `private` endpoint?

Comment (gssbzn) [reply internal/validate/common_password.go:17]: same as we did with with current IP

Comment (gssbzn) [reply internal/validate/common_password.go:17]: > Instead of looking up for weak password, could we add a password policy?

Check for known common passwords is a good strategy, password policies are actually a bad practice now days, length and uniqueness (across services) is all that matters

File: internal/validate/common_password.go
Comment (gssbzn) [inline internal/validate/common_password.go:52]: @andreaangiolillo 👀

File: internal/validate/common_password.go
Comment (andreaangiolillo) [reply internal/validate/common_password.go:17]: yes, I was thinking about the private endpoint but I wasn’t sure it was worth it

Comment (andreaangiolillo) [reply internal/validate/common_password.go:17]: @threebee this is the endpoint

`Error: POST https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/databaseUsers: 400 (request "COMMON_PASSWORD") The password provided is too weak, as it can be found in most commonly used password lists.`

Comment (gssbzn) [reply internal/validate/common_password.go:17]: > yes, I was thinking about the private endpoint but I wasn’t sure it was worth it

I think it is but I would not block this PR on that change as we may want to discuss it further with other teams

File: internal/validate/common_password.go
Comment (andreaangiolillo) [reply internal/validate/common_password.go:52]: ahahaha I didn’t notice that xD

File: internal/validate/validate.go
Comment (andreaangiolillo) [reply internal/validate/validate.go:167]: added!

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:175]: something I just realised the map is all lowercase you may want to do this instead to be safe
```suggestion
	if commonPasswords[strings.ToLower(password)] {
		return ErrWeakPassword
	}
```

File: internal/validate/common_password.go
Comment (andreaangiolillo) [reply internal/validate/common_password.go:17]: I am going to create a new ticket to discuss this private endpoint once I merge this

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:177]: check length before, some of those common password are shorter and may fail here first

File: internal/validate/validate.go
Comment (gssbzn) [inline internal/validate/validate.go:178]: based on previous comment on formatting
```suggestion
		return fmt.Erorf("%w min: %d", ErrShortPassword, minPasswordLength)
```

**Review by gssbzn — COMMENTED** [review]
almost there

**Review by gssbzn — APPROVED** [review]
LGTM :shipit:

**Pull request merged**
